### PR TITLE
move some scripts from 'build' to 'tools' directory, misc refactors

### PR DESCRIPTION
**Pull request by silverwind**
- Move some scripts from `build` to new `tools` dir. Eventually i would like to move all but let's do it step-by-step.
- Add dir to eslint and move the files into vars.
- Update docs accordingly.
- While updating docs I noticed we were incorrectly having `public/img` path still in a few places. Replace those with the current `public/assets/img`.

### CODE REVIEW

File: docs/content/development/hacking-on-gitea.zh-cn.md [hunk 1]
```diff
@@ -201,7 +201,7 @@ REPO_INDEXER_CONN_STR = http://elastic:changeme@localhost:9200
 
 ### 构建和添加 SVGs
 
-SVG 图标是使用 `make svg` 目标构建的，该目标将 `build/generate-svg.js` 中定义的图标源编译到输出目录 `public/img/svg` 中。可以在 `web_src/svg` 目录中添加自定义图标。
+SVG 图标是使用 `make svg` 目标构建的，该目标将图标源编译到输出目录 `public/assets/img/svg`。可以在 `web_src/svg` 目录中添加自定义图标。
```
Comment (silverwind) [inline docs/content/development/hacking-on-gitea.zh-cn.md]: @lunny please check this one, it's partially machine-translated.

File: .dockerignore [hunk 2]
```diff
@@ -62,7 +62,7 @@ cpu.out
 /data
 /indexers
 /log
-/public/img/avatar
+
```
Comment (silverwind) [inline .dockerignore]:
```suggestion
```

**Comment by denyskon** [comment]
Wait, why do we still have Chinese docs? Didn't we decide to only have English, or am I hallucinating?

**Comment by silverwind** [comment]
Not sure about such a decision, but I would be all for removing chinese docs.

As this PR proves again, they are just becoming a outdated mess over time and hold up review process because there is really only one active chinese speaker in the reviewers.

File: docs/content/development/hacking-on-gitea.zh-cn.md (hunk 1 above)
Comment (nanguanlin6) [inline docs/content/development/hacking-on-gitea.zh-cn.md]: use this one below
SVG 图标是使用 `make svg` 命令构建的，该命令将 `build/generate-svg.js` 中定义的图标资源编译到输出目录 `public/assets/img/svg` 中。可以在 `web_src/svg` 目录中添加自定义图标。

**Review by nanguanlin6 — APPROVED** [review]

File: docs/content/development/hacking-on-gitea.zh-cn.md (hunk 1 above)
Comment (nanguanlin6) [reply docs/content/development/hacking-on-gitea.zh-cn.md]: Sorry. Accidentally clicked `approve`.

Comment (silverwind) [inline docs/content/development/hacking-on-gitea.zh-cn.md]:
```suggestion
SVG 图标是使用 make svg 命令构建的，该命令将 build/generate-svg.js 中定义的图标资源编译到输出目录 public/assets/img/svg 中。可以在 web_src/svg 目录中添加自定义图标。
```

Comment (silverwind) [reply docs/content/development/hacking-on-gitea.zh-cn.md]: Added in https://github.com/go-gitea/gitea/pull/29844/commits/5d907045b09f0b9a4c984307fd50e05a5236c003. Next time, just use the suggestion feature.

File: docs/content/development/hacking-on-gitea.zh-cn.md [hunk 3]
```diff
@@ -201,7 +201,7 @@ REPO_INDEXER_CONN_STR = http://elastic:changeme@localhost:9200
 
 ### 构建和添加 SVGs
 
-SVG 图标是使用 `make svg` 目标构建的，该目标将 `build/generate-svg.js` 中定义的图标源编译到输出目录 `public/img/svg` 中。可以在 `web_src/svg` 目录中添加自定义图标。
+SVG 图标是使用 make svg 命令构建的，该命令将 build/generate-svg.js 中定义的图标资源编译到输出目录 public/assets/img/svg 中。可以在 web_src/svg 目录中添加自定义图标。
```
Comment (silverwind) [inline docs/content/development/hacking-on-gitea.zh-cn.md]: @lng2020 actually, it's still not correct. The change in english is only about removing the "defined in `build/generate-svg.js`" phrase, but it's still present here. Could you make another suggestion?

Comment (nanguanlin6) [inline docs/content/development/hacking-on-gitea.zh-cn.md]:
```suggestion
SVG 图标是使用 `make svg` 命令构建的，该命令将图标资源编译到输出目录 `public/assets/img/svg` 中。可以在 `web_src/svg` 目录中添加自定义图标。
```

File: docs/content/administration/cmd-embedded.zh-cn.md [hunk 4]
```diff
@@ -37,7 +37,7 @@ gitea embedded list [--include-vendored] [patterns...]
 
 - 列出所有模板文件，无论在哪个虚拟目录下：`**.tmpl`
 - 列出所有邮件模板文件：`templates/mail/**.tmpl`
-- 列出 `public/img` 目录下的所有文件：`public/img/**`
+- 列出 `public/assets/img` 目录下的所有文件：`public/asset/img/**`
```
Comment (nanguanlin6) [inline docs/content/administration/cmd-embedded.zh-cn.md]:
```suggestion
列出 `public/assets/img` 目录下的所有文件：`public/assets/img/**`
```

**Comment by wxiaoguang** [comment]
> Move some scripts from `build` to new `tools` dir. Eventually i would like to move all but let's do it step-by-step.

Do you mean to rename "build" to "tools" eventually? TBH I don't see real benefit or difference.

**Comment by techknowlogick** [comment]
> Do you mean to rename "build" to "tools" eventually? TBH I don't see real benefit or difference.

I see it as `tools` is less specific to building gitea itself, as with generating the logo can be done with Gitea already being built.

**Comment by wxiaoguang** [comment]
> as with generating the logo can be done with Gitea already being built.

Yup, while I think "build/generate-svg.js" could also be interpreted as "building Gitea related assets".

Actually I am neutral for this renaming. My major question is `Do you mean to rename "build" to "tools" eventually?`

* If yes: some files like "test-env-check.sh" doesn't seems to be a "tool"
* If no: what kind of files should be left in "build" while what kind of files should be moved to "tools"? A plan would be very helpful.

**Comment by silverwind** [comment]
> Do you mean to rename "build" to "tools" eventually?

Yes. I find the "build" name unsuitable for general scripts. For all I know "build" could also contain built files, which it doesn't.

> If yes: some files like "test-env-check.sh" doesn't seems to be a "tool" (maybe they need a new directory?)

As I see it, every script is a tool.

**Review by wxiaoguang — APPROVED** [review]
Hopefully "build" and "tools" directories won't co-exist for long time, otherwise people might hesitate about which one to use 😄

**Pull request merged**